src/descriptors/power_system_structs.json

```diff
@@ -132,6 +126,13 @@
           },
           "validation_action": "warn"
         },
+        {
+          "name": "ancillary_services",
+          "comment": "Bids for the ancillary services",
+          "null_value": "Dict{String, Float64}()",
+          "data_type": "Dict{String, Float64}",
```

Comment:
What is the idea behind this dict exactly? Is it something like

```
"regulation" => C^{reg}
"spinning" => C^{spin}
...
```

or maybe having the strings as the names of the services?

Reply:
the strings should match the names of the services. We will probably want something that can do this automatically when running add service but this will come later as we refine the interface.

Reply:
In that case, is there any reason not to have the dict as a `Dict{Service, Float64}`?

Reply:
The service structs are mutable, so they should not be keys of a dictionary.  You could use `IdDict`, however.

If (1) you need to be able to find the services objects and (2) the services can be any subtype of `Service` then storing only the name is technically not sufficient for looking up a service with `get_component`.  You need the concrete type as well.  That's why we have `ServiceContributingDevicesKey`.  I may not have known about `IdDict` when I implemented that.

If these objects will be stored in the `System` and you do include the actual service objects then some customization will be needed for serialization.  Keys of JSON objects must be strings.  We would have to substitute their UUIDs.

Reply:
@daniel-thom if the services are already stored in the same device, this should be simpler. The search would go in the other direction using the service in the device to find the bid.

Reply:
OK, that makes sense.  Are you still vulnerable to a name collision?  Services of different concrete types could have the same name.  All code that adds to this dict could assert uniqueness as a guarantee.

Reply:
Given that dicts don't allow for duplicate keys it will require a sequence of user mistakes to end up with such a situation.

Reply:
What would be the user mistakes?  In your comment above you said that we will eventually have code that automatically builds this dictionary.  Users are allowed to have duplicate names across different concrete types.  So, they could end up with a collision and not know it.

Reply:
The error will entail creating the cost function with the names of the services first and then adding to the device 2 services with the exact same name after. 

To avoid this one option is to do `add_service!()` and pass the bid and recorded in the cost function. The other option is that users add the key to the dict when making the cost function but we won't be able to guarantee there is a service with that name.

Reply:
Even if this is all very unlikely I think we should keep the design consistent.  The design allows a device to have services of differing types.  Therefore, it allows a device to have services with the same name as long as the types are different.  We could add restrictions to this, of course.

If this dictionary will be an internally-managed field (never created or accessed by the user) then you could use the UUID instead of the name.

Reply:
As discussed, we will go with `IdDicts` such that the key to the bid is the service in questions. Please note that this will require creating the component and the service BEFORE been able to add the bids. As a result, by default, the bids dict will always start empty.